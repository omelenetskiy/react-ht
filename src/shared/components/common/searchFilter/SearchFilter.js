import React from 'react';

import StyledSearchFilter from '../../styled/searchFilter';
import Button from '../button/Button';
import ButtonGroup from '../buttonGroup/ButtonGroup';

const SearchFilter = ({ handleSearch }) => (
  <StyledSearchFilter>
    <ButtonGroup />
    <Button size="large" onClick={handleSearch}>
      Search
    </Button>
  </StyledSearchFilter>
);

export default SearchFilter;
