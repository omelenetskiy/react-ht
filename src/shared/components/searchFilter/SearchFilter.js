import React, { Component } from 'react';
import { connect } from 'react-redux';
import { actionSearchByChange } from '../../store/actions/movies';

import { searchByButtons } from '../../../config/buttonGroups';

import StyledSearchFilter from '../../styled/searchFilter';
import Button from '../button/Button';
import ButtonGroup from '../buttonGroup/ButtonGroup';

class SearchFilter extends Component {
  toggleSearchBy = (e) => {
    const { changeSearchBy } = this.props;
    changeSearchBy(e.target.value);
  };

  render() {
    const { handleSearch } = this.props;
    return (
      <StyledSearchFilter>
        <ButtonGroup
          controls={searchByButtons}
          toggleSearchBy={this.toggleSearchBy}
          color="light"
        />
        <Button size="large" onClick={handleSearch}>
          Search
        </Button>
      </StyledSearchFilter>
    );
  }
}

const matDispatchToProps = (dispatch) => ({
  changeSearchBy: (value) => dispatch(actionSearchByChange(value)),
});

export default connect(
  null,
  matDispatchToProps
)(SearchFilter);
