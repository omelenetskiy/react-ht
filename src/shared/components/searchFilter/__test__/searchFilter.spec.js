import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-enzyme';
import 'jest-styled-components';

import SearchFilter from '../SearchFilter';

const handleChangeSearchBy = jest.fn();
const handleSearch = jest.fn();

const props = {
  handleChangeSearchBy,
  handleSearch,
};

describe('SearchFilter', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<SearchFilter {...props} />);
    wrapper.instance().toggleSearchBy({ target: { value: 'cats' } });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
