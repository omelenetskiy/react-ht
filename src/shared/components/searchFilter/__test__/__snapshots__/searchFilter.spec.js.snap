// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilter should render correctly 1`] = `
<searchFilter___default>
  <ButtonGroup
    color="light"
    controls={
      Object {
        "buttons": Array [
          Object {
            "checked": true,
            "name": "search",
            "title": "Title",
            "value": "title",
          },
          Object {
            "checked": false,
            "name": "search",
            "title": "Genres",
            "value": "genres",
          },
        ],
        "title": "SEARCH BY",
      }
    }
    toggleSearchBy={[Function]}
  />
  <Button
    onClick={[MockFunction]}
    size="large"
  >
    Search
  </Button>
</searchFilter___default>
`;
